using-mobile-extensions/adobe-edge-identity/edge-identity-api-reference.md

```diff
@@ -0,0 +1,520 @@
+# Edge Identity API Reference
+
+## getExperienceCloudId
+
+{% tabs %}
+{% tab title="Android" %}
+### getExperienceCloudId
+
+This API retrieves the ECID that was generated when the app was initially launched.
+
+This ID is preserved between app upgrades, is saved and restored during the standard application backup process, and is removed at uninstall. The values are returned via the [AdobeCallback](../mobile-core/mobile-core-api-reference.md#public-classes).
+
+When [AdobeCallbackWithError](../mobile-core/mobile-core-api-reference.md#public-classes) is provided, and you are fetching the ECID from the Mobile SDK, the timeout value is 500ms. If the operation times out or an unexpected error occurs, the `fail` method is called with the appropriate [AdobeError](../mobile-core/mobile-core-api-reference.md#public-classes).
+
+**Java**
+
+**Syntax**
+
+```java
+public static void getExperienceCloudId(final AdobeCallback<String> callback);
+```
+
+* _callback_ is invoked after the ECID is available.
+
+**Example**
+
+```java
+Identity.getExperienceCloudId(new AdobeCallback<String>() {    
+    @Override    
+    public void call(String id) {        
+         //Handle the ID returned here    
+    }
+});
+```
+{% endtab %}
+
+{% tab title="iOS" %}
+### getExperienceCloudId
+
+This API retrieves the ECID that was generated when the app was initially launched.
+
+This ID is preserved between app upgrades, is saved and restored during the standard application backup process, and is removed at uninstall. The values are returned via the callback.
+
+**Syntax**
+
+```swift
+@objc(getExperienceCloudId:)
+static func getExperienceCloudId(completion: @escaping (String?, Error?) -> Void)
+```
+
+* _completion_ is invoked after the ECID is available.  The default timeout is 1000ms.
+
+**Examples**
+
+**Objective-C**
+
+```objectivec
+[AEPEdgeIdentity getExperienceCloudId:^(NSString * _Nullable ecid, NSError * error) {   
+    // handle the retrieved ID here    
+}];
+```
+
+**Swift**
+
+```swift
+Identity.getExperienceCloudId { (ecid, error) in
+  if let error = error {
+    // handle error here
+  } else {
```

Comment:
`ecid` can still be nil here. Should we also unwrap `ecid`?

`} else if let ecid = ecid { ... }`

Reply:
This example is to show the user should check if an error exists before using the returned `ecid` value, with just a generic statement of "handle the retrieved identifiers here" if there is no error. I don't want to bog down the example with too many checks. The syntax example does state they value is optional. Conversely, I could just go with,
```
Identity.getExperienceCloudId { (ecid, error) in
   // handle ecid 
}
```